controllers/component_build_controller.go

```diff
@@ -485,6 +491,9 @@ func (r *ComponentBuildReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 
 		// Update build status annotation
 		buildStatus := readBuildStatus((&component))
+		if pacBuildStatus.ConfigurationTime == "" && buildStatus.PaC != nil && buildStatus.PaC.ConfigurationTime != "" {
```

Comment:
not sure if this is needed for unconfigure. That means that configuration-time will be time of when it was unconfigured?

Reply:
The way I understand it:

Successfully configure PaC -> `BuildStatus` gets `ConfigurationTime`

Unconfigure PaC -> overwrite whole `buildStatus.PaC` by newly created `pacBuildStatus` which does not have set `ConfigurationTime` attribute, so the previously set `ConfigurationTime` value would be lost and would be empty, which is not intended behavior.

Here, I get the previously set `ConfigurationTime` and give it to the `pacBuildStatus` so it does not get lost

I assume that attribute `ConfigurationTime` becomes empty when the `buildStatus.PaC` is assigned to `pacBuildStatus` on line 497 which does not have that attribute, like in [this example](https://go.dev/play/p/n6cuQ3IYeh8)

Reply:
So that means that in next configuration the time will not be updated because it's already set. Our goal is to have configuration time updated when new configuration is created.
For me it's ok to keep ConfigurationTime not after unconfiguraton.

Reply:
There is a condition in the if statement checking that configuration time did not get changed during latest run `pacBuildStatus.ConfigurationTime == ""`
That means it should change the configuration time only if it was not already changed [here on line 415](https://github.com/redhat-appstudio/build-service/pull/181/files/a9800fb2e718b88880e3954b2ae68137082cda32#diff-55dabc319dc9f0a58a4c79d69d79368078f475199412ea72d3fae8bd7c659785R415), shouldn't it?

I see that this code seems confusing, I will try to think of some way to improve it